src/lib/core/fiber_channel.h

```diff
@@ -137,7 +137,15 @@ struct fiber_channel {
 	struct rlist waiters;
 	/** Ring buffer read position. */
 	uint32_t beg;
-	/* True if the channel is closed. */
+	/**
+	 * True if the channel forbids both reading and writing and
+	 * the buffer is no longer accessible.
+	 */
+	bool is_destroyed;
+	/**
+	 * True if the channel is closed for writing and
+	 * is waiting to be destroyed or has been destroyed.
+	 */
 	bool is_closed;
```

Comment:
I would reorder the flags: the channel is closed first, then destroyed. The order opposite to one present now would be more natural.

Reply:
Done.

Reply:
(And in `fiber_channel_create()`. Feel free to ignore.)

Reply:
Also fixed.